### PR TITLE
fix the requirements check in launch_cluster script

### PR DESCRIPTION
**Pull request by acezen**
fix the requirements check in launch_cluster script and try to create directory before output config file.

### CODE REVIEW

File: scripts/launch_cluster.py [hunk 1]
```diff
@@ -46,23 +46,39 @@
...
-def check_file_exist(file):
-    if os.path.isfile(file):
+def check_requirements(cloud_type, file_path):
+    # check output file is existed
+    ready = True
+    click.echo("Checking requirements...")
+    if os.path.isfile(file_path):
         click.echo(
-            "%s exist already, please rmove or backup it and run script again." % file
+            "* %s exist already, please rmove or backup it." % file_path
```
Comment (siyuan0322) [inline scripts/launch_cluster.py:55]: already exists;   rmove -> remove;   make a backup.

File: scripts/launch_cluster.py [hunk 2]
```diff
@@ -684,22 +702,21 @@ def launch(type, id, secret, region, output):
...
-    This script assumes that kubectl and aws-iam-authenticator are in your PATH.
+    This script assumes that kubectl in your PATH and if use AWS cluster, you also need to
+    install aws-iam-authenticator.
     """
 
-    check_file_exist(output)
-    check_kubectl_installed()
+    if not check_requirements(type, output):
+        click.echo(
+            "Requirements of the script not ready, please follow the promp to install requirement."
```
Comment (siyuan0322) [inline scripts/launch_cluster.py:711]: promp -> prompt

File: scripts/launch_cluster.py [hunk 3]
```diff
@@ -462,6 +478,7 @@ def _write_kube_config(self, cluster_name):
 
         # Write in YAML.
         config_text = yaml.dump(cluster_config, default_flow_style=False)
+        os.makedirs(os.path.dirname(self._output_path), exist_ok=True)
```
Comment (siyuan0322) [inline scripts/launch_cluster.py:481]: I remerber that you also need to write a `aws-auth-cm.yaml`? That file also need to create its directory first.

File: scripts/launch_cluster.py [hunk 4]
```diff
@@ -684,22 +702,21 @@ def launch(type, id, secret, region, output):
...
-            )
         launcher = AWSLauncher(id, secret, region, output)
     elif type == "aliyun":
-        if CS20151215Client is None:
-            click.echo(
-                "aliyun sdk not found, please install by \"pip3 install alibabacloud_cs20151215 alibabacloud_ecs20140526 alibabacloud_vpc20160428\"."
-            )
         launcher = AliyunLauncher(id, secret, region, output)
     else:
         click.echo("Not support cloud type %s" % type)
```
Comment (siyuan0322) [inline scripts/launch_cluster.py:722]: Still proceed to launch cluster if  unsupported cloud type is met.
"Not support" -> "Not supported"?

Comment (acezen) [reply scripts/launch_cluster.py:722]: actually， the option can only be aws or aliyun, here will  remove the else branch

**Review by siyuan0322 — APPROVED** [review]

**Pull request merged**